src/clappr-dash-shaka-playback.js

```diff
@@ -140,19 +150,42 @@ class DashShakaPlayback extends HTML5Video {
     return (this.isReady && this._player.isLive() ? 'live' : 'vod') || ''
   }
 
-  selectTrack (track) {
-    if (track.type === 'text') {
-        this._player.selectTextTrack(track)
-    } else if (track.type === 'variant') {
-        this._player.selectVariantTrack(track)
-        if (track.mimeType.startsWith('video/')) {
-            // we trigger the adaptation event here
-            // because Shaka doesn't trigger its event on "manual" selection.
-            this._onAdaptation()
-        }
-    } else {
-        throw new Error('Unhandled track type:', track.type);
+  selectTrack (track, clearBuffer) {
+    if (this.isReady) {
+      return false
+    }
```

Comment:
Should we check if `isReady` or `!isReady`?

Reply:
nice catch! didn't test selecting variant tracks here again. thanks!!

my next aim on this codebase is to add some unit tests :)